### PR TITLE
Tests for waf collection source type

### PR DESCRIPTION
**Pull request by avdata99**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![CircleCI](https://circleci.com/gh/datopian/ckanext-geodatagov.svg?style=svg)](https://circleci.com/gh/datopian/ckanext-geodatagov)
```
Comment (adborden) [inline README.md]: Should be GSA repo.

File: ckanext/geodatagov/harvesters/base.py [hunk 2]
```diff
@@ -15,7 +17,7 @@
 from ckanext.spatial.harvesters.base import guess_standard
 
 from ckanext.spatial.validation import Validators
-
+# from ckanext.spatial.validation import ISO19139NGDCSchema
```
Comment (adborden) [inline ckanext/geodatagov/harvesters/base.py]: Is this comment necessary?

File: ckanext/geodatagov/harvesters/base.py [hunk 3]
```diff
@@ -48,13 +58,16 @@ def default_groups_validator(value):
...
+
 class GeoDataGovHarvester(SpatialHarvester):
 
     def extra_schema(self):
+        log.debug('Getting extra schema for GeoDataGovHarvester')
         return {
             'private_datasets': [ignore_empty, boolean_validator],
             'default_groups': [ignore_empty, default_groups_validator, lambda value: [value]],
-            'validator_profiles': [ignore_empty, unicode, validate_profiles, lambda value: [value]],
+            # 'validator_profiles': [ignore_empty, unicode, validate_profiles, lambda value: [value]],
```
Comment (adborden) [inline ckanext/geodatagov/harvesters/base.py]: I think this comment can be removed.

File: ckanext/geodatagov/harvesters/base.py [hunk 4]
```diff
@@ -105,18 +119,22 @@ def transform_to_iso(self, original_document, original_format, harvest_object):
             return None
 
         # Validate against FGDC schema
+
         if self.source_config.get('validator_profiles'):
             profiles = self.source_config.get('validator_profiles')
         else:
             profiles = ['fgdc_minimal']
+        log.debug('Validatos profiles for transform iso {}'.format(profiles))
```
Comment (adborden) [inline ckanext/geodatagov/harvesters/base.py]: s/Validatos/Validates/

File: pip-requirements.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
 -e git+https://github.com/asl2/PyZ3950.git#egg=py3250
 ply==3.4
-boto
\ No newline at end of file
+boto
```
Comment (adborden) [inline pip-requirements.txt]: Why boto?

File: ckanext/geodatagov/harvesters/base.py (hunk 2 above)
Comment (avdata99) [reply ckanext/geodatagov/harvesters/base.py]: No, it's not. I just deleted

File: ckanext/geodatagov/harvesters/base.py (hunk 3 above)
Comment (avdata99) [reply ckanext/geodatagov/harvesters/base.py]: deleted

File: ckanext/geodatagov/harvesters/base.py (hunk 4 above)
Comment (avdata99) [reply ckanext/geodatagov/harvesters/base.py]: Fixed

File: pip-requirements.txt (hunk 5 above)
Comment (avdata99) [reply pip-requirements.txt]: Is for [getting data from Amazon S3](https://github.com/datopian/ckanext-geodatagov/blob/master/ckanext/geodatagov/commands.py#L1328)

Comment (adborden) [reply pip-requirements.txt]: Yes, but why are you adding it? I don't see any other references to boto or S3 in your diff. Is the dependency missing?

Comment (avdata99) [reply pip-requirements.txt]: I think I missed this line. Was added [here](https://github.com/datopian/ckanext-geodatagov/commit/f0afe1d6b102ad398df504676dfb160a626791dc). The requirement and the command that uses it were added at the same time.

File: README.md (hunk 1 above)
Comment (avdata99) [reply README.md]: fixed [here](https://github.com/datopian/ckanext-geodatagov/commit/41307320db41d9285fa4f8f56b1cf70d3d356546)

**Comment by avdata99** [comment]
All the tests are passing again in CIrcleCI. I did some minor fixes.

**Comment by avdata99** [comment]
> Can you add instructions to the README on how to run the tests?

Added some notes [here](https://github.com/datopian/ckanext-geodatagov/commit/f0103d5838aaafd89764210a063533092e83ecd0) @adborden

File: bin/travis-build.bash [hunk 6]
```diff
@@ -7,17 +7,16 @@ echo "Installing the packages that CKAN requires..."
...
-pip install setuptools -U
-
 # if [ $CKANVERSION == '2.8' ]
 # then
-# 	sudo apt-get install postgresql-9.6
+# 	sudo apt-get install postgresql-9.3-postgis-2.1
 # elif [ $CKANVERSION == '2.3' ]
 # then
-# 	sudo apt-get install postgresql-9.1
+# 	sudo apt-get install postgresql-9.1-postgis
```
Comment (adborden) [inline bin/travis-build.bash]: Is the commented code necessary?

File: README.md [hunk 7]
```diff
@@ -1,3 +1,5 @@
+[![CircleCI](https://circleci.com/gh/gsa/ckanext-geodatagov.svg?style=svg)](https://circleci.com/gh/datopian/ckanext-geodatagov)
```
Comment (adborden) [inline README.md]:
```suggestion
[![CircleCI](https://circleci.com/gh/gsa/ckanext-geodatagov.svg?style=svg)](https://circleci.com/gh/gsa/ckanext-geodatagov)
```

File: bin/travis-build.bash [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+cd .. # CircleCI starts inside ckanext-geodatagov folder
+pwd
+ls -la
+
+git clone https://github.com/ckan/ckan
+cd ckan
+if [ $CKANVERSION == '2.8' ]
+then
+	git checkout 2.8
+elif [ $CKANVERSION == '2.3' ]
```
Comment (adborden) [inline bin/travis-build.bash]: For 2.3, we actually want the data.gov fork, https://github.com/GSA/ckan/tree/datagov

File: bin/travis-build.bash [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+echo "-----------------------------------------------------------------"
+echo "Initialising the database..."
+cd ckan
+paster db init -c test-core.ini
+
+cd ..
+echo "-----------------------------------------------------------------"
+echo "Installing Harvester"
+git clone https://github.com/ckan/ckanext-harvest
```
Comment (adborden) [inline bin/travis-build.bash]: Should this be the GSA fork? https://github.com/GSA/ckanext-harvest/tree/datagov

File: ckanext/geodatagov/harvesters/base.py [hunk 10]
```diff
@@ -35,9 +36,17 @@
 
 
 def validate_profiles(profile):
-    if profile not in [p[0] for p in VALIDATION_PROFILES]:
+    if isinstance(profile, list) and len(profile) == 1:
+        profile = profile[0]
+    validation_profile_names = [p[0] for p in VALIDATION_PROFILES]
+    log.info('Validate {} ({}) in profiles func {}'.format(profile, type(profile), validation_profile_names))
```
Comment (adborden) [inline ckanext/geodatagov/harvesters/base.py]: This should be a debug log.

File: bin/travis-build.bash (hunk 6 above)
Comment (avdata99) [reply bin/travis-build.bash]: removed

File: bin/travis-build.bash (hunk 8 above)
Comment (avdata99) [reply bin/travis-build.bash]: fixed

File: bin/travis-build.bash (hunk 9 above)
Comment (avdata99) [reply bin/travis-build.bash]: Fixed

File: ckanext/geodatagov/harvesters/base.py (hunk 10 above)
Comment (avdata99) [reply ckanext/geodatagov/harvesters/base.py]: fixed

**Comment by avdata99** [comment]
Since the PR accepted on ckanex-harvest, now the tests here are passing @adborden

**Pull request merged**